src/helper/utils.js

```diff
@@ -0,0 +1,190 @@
+ *     merged onto the dataLayer's model.
+ * @package
+ */
+function expandKeyValue(key, value) {
+  const result = {};
+  let target = result;
+  const split = key.split('.');
+  for (let i = 0; i < split.length - 1; i++) {
+    target = target[split[i]] = {};
+  }
+  target[split[split.length - 1]] = value;
+  return result;
+}
+
+/**
+ * Determines if the given value is an array.
+ *
+ * @param {*} value The value to test.
+ * @return {boolean} True iff the given value is an array.
+ * @package
+ */
+function isArray(value) {
+  return type(value) === 'array';
+}
+
+/**
+ * Determines if the given value is an arguments object.
+ *
+ * @param {*} value The value to test.
+ * @return {boolean} True iff the given value is an arguments object.
+ * @package
+ */
+function isArguments(value) {
+  return type(value) === 'arguments';
+}
+
+/**
+ * Determines if the given value is a string.
+ *
+ * @param {*} value The value to test.
+ * @return {boolean} True iff the given value is a string.
+ * @package
+ */
+function isString(value) {
+  return type(value) === 'string';
+}
+
+/**
+ * Merges one object into another or one array into another. Scalars and
+ * "non-plain" objects are overwritten when there is a merge conflict.
+ * Arrays and "plain" objects are merged recursively.
+ *
+ * TODO(bkuhn): This is just a starting point for how we'll decide which
+ * objects get cloned and which get copied. More work is needed to flesh
+ * out the details here.
+ *
+ * @param {!Array<*>|!Object<*>} from The object or array to merge from.
+ * @param {!Array<*>|!Object<*>} to The object or array to merge into.
+ * @package
+ */
+function merge(from, to) {
+  const allowMerge = !from['_clear'];
+  for (const property in from) {
+    if (hasOwn(from, property)) {
+      const fromProperty = from[property];
+      if (isArray(fromProperty) && allowMerge) {
+        if (!isArray(to[property])) to[property] = [];
+        merge(fromProperty, to[property]);
+      } else if (isPlainObject(fromProperty) && allowMerge) {
+        if (!isPlainObject(to[property])) to[property] = {};
+        merge(fromProperty, to[property]);
+      } else {
+        to[property] = fromProperty;
+      }
+    }
+  }
+  delete to['_clear'];
+}
+
+
+/**
+ * @define {boolean} When true, potential code errors will be logged to the
+ * console. To enable this, run yarn build-debug to compile the distribution
+ * code.
+ */
+const DLH_DEBUG = goog.define('DLH_DEBUG', false);
+
+/**
+ * Enum for choosing the level at which to log an error.
+ * @readonly
+ * @enum {number}
+ * @package
+ */
+const LogLevel = {
+  LOG: 1,
+  WARNING: 2,
+  ERROR: 3,
+};
+
+/**
+ * Log an error to the console if the debug distribution is in use.
+ *
+ * @param {string} toLog The error to log to the console in debug mode.
+ * @param {!LogLevel} logLevel The error to log to the console in debug mode.
+ * @package
+ */
+function logError(toLog, logLevel) {
```

Comment:
Since we have info/warning here, I'd just call this function "log".

Reply:
Sounds good! It probably won't be confused with the math function since it takes a string argument